src/Processors/Transforms/Streaming/AggregatingTransform.h

```diff
@@ -51,6 +59,8 @@ struct AggregatingTransformParams
         return res;
     }
 
+    bool repeatEmit() const noexcept { return emit_repeat && !emit_changelog && emit_mode < EmitMode::OnUpdate; }
```

Comment:
I think we don't need it.
Since the global aggregation already checked `!emit_changelog` and the AST parsing ensured `emit_mode` must be emit periodic as well

Reply:
Why global aggr cannot emit changelog ?